internal/store/store.go

```diff
@@ -0,0 +1,134 @@
+package store
+
+import (
+	"database/sql"
+	"net/url"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	// enable the pq driver
+	_ "github.com/lib/pq"
+	// enable the sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// SQLStore abstracts access to the database.
+type SQLStore struct {
+	db     *sqlx.DB
+	logger logrus.FieldLogger
+}
+
+// New constructs a new instance of SQLStore.
+func New(dsn string, logger logrus.FieldLogger) (*SQLStore, error) {
+	url, err := url.Parse(dsn)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse dsn as an url")
+	}
+
+	var db *sqlx.DB
+
+	switch strings.ToLower(url.Scheme) {
+	case "sqlite", "sqlite3":
+		db, err = sqlx.Connect("sqlite3", url.Host)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to connect to sqlite database")
+		}
+
+		// Override the default mapper to use the field names "as-is"
+		db.MapperFunc(func(s string) string { return s })
+
+	case "postgres", "postgresql":
+		url.Scheme = "postgres"
+		db, err = sqlx.Connect("postgres", url.String())
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to connect to postgres database")
+		}
+
+		// Leave the default mapper as strings.ToLower.
+
+	default:
+		return nil, errors.Errorf("unsupported dsn scheme %s", url.Scheme)
+	}
+
+	return &SQLStore{
+		db,
+		logger,
+	}, nil
+}
+
+// queryer is an interface describing a resource that can query.
```

Comment:
The following interfaces and functions are the "thin layer" I spoke of in the PR's description. At first, I intended to fully adopt `sqlx.*` types throughout the store, but I ran into a few limitations:
* `Exec`, by default, doesn't rebind the `?` placeholder for the database in question. There's an easy `Rebind` that does this, but it's annoying to do at the caller.
* There's a handy `NamedExec` that both rebinds and accepts structs and maps, but there's no top-level `NamedGet` equivalent. You can prepare a `NamedStmt` to get the `NamedGet` equivalent, but once again this is annoying at the caller.
* I wanted to leverage squirrel for those times when the SQL needs to be dynamic, but it's slightly annoying to have to extract the sql and args only then to pass them back into sqlx.

All of this spoke to the need for a thin layer to simplify usage in the store. I realized quickly that I could constrain `sqlx` entirely to this file, and so added a few more helper functions accordingly.